publication-commons/src/test/java/no/unit/nva/doi/models/ImmutableDoiTest.java

```diff
@@ -0,0 +1,294 @@
+package no.unit.nva.doi.models;
+
+import static no.unit.nva.doi.models.ImmutableDoi.ERROR_DOI_URI_INVALID_FORMAT;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.Matchers.sameInstance;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import java.net.URI;
+import java.util.UUID;
+import org.junit.jupiter.api.Test;
+
+class ImmutableDoiTest {
+
+    public static final URI DOI_PROXY = URI.create("https://doi.org/");
+    public static final String EXAMPLE_SUFFIX = createRandomSuffix();
+    public static final String EXAMPLE_RANDOM_VALUE = "2";
+    public static final String FORWARD_SLASH = "/";
+    public static final String REQUIRED_ATTRIBUTES_ARE_NOT_SET = "required attributes are not set";
+    public static final String ERROR_STRICT_BUILDER = "Builder of Doi is strict, attribute is already set";
+    public static final String URI_VALID_EMAILTO_BUT_INVALID_URL = "emailto:nope@example.net";
+    private static final URI STAGE_DOI_PROXY = URI.create("https://handle.stage.datacite.org/");
+    private static final String DEMO_PREFIX = "10.5072";
+    public static final String EXAMPLE_PREFIX = DEMO_PREFIX;
+    public static final String EXAMPLE_IDENTIFIER = EXAMPLE_PREFIX + FORWARD_SLASH + EXAMPLE_SUFFIX;
+    private static final URI EXAMPLE_PROXY = STAGE_DOI_PROXY;
+    public static final URI EXAMPLE_DOI = URI.create(EXAMPLE_PROXY + EXAMPLE_IDENTIFIER);
+    private static final String EXAMPLE_PREFIX_2 = "10.16903";
+
+    @Test
```

Comment:
Forventer at blant annet, følgende ikke valideres som DOI:

`            "https://example.org/local.irrigation/burlapSack",
            "https://example.org/wanderlust/freakout",
            "https://doi.org/11.9821/corgismack",
            "ftp://doi.org/murkyWater/fretwork"`

Og at withIdentifier ikke aksepterer disse som verdier.

Reply:
https://github.com/BIBSYSDEV/nva-publication-api/pull/96/commits/2fe80f388dc107333cfa18ae88b4273e0c03c374 and https://github.com/BIBSYSDEV/nva-publication-api/pull/96/commits/3e033e023c96cf3329d2cfe4ccb8c2ac758e73d8 covers it?

Reply:
Together with cfbf0f01cdabe7aa0d1f3064a9c6ff475949aa8a I believe.

Reply:
Also added https://github.com/BIBSYSDEV/nva-publication-api/pull/96/commits/004083546b6399a149f6429c272d0d6b2cef39e4 to test for all known valid proxies and schemes.

Reply:
Lets add more validation if needed in the `Builder#build` if we notice we need more validation: https://github.com/BIBSYSDEV/nva-publication-api/pull/96#issuecomment-731022543